src/spn/algorithms/layerwise/distributions.py

```diff
@@ -0,0 +1,465 @@
+"""
+Module that contains a set of distributions with learnable parameters.
+"""
+from abc import ABC
```

Comment:
What is ABC here?

Reply:
Its Python's `AbstractBaseClass` to ensure proper inheritance for the `Leaf` class and its subclasses.